### PR TITLE
Middle Man with a few qubits, documentation for superdense coding, te…

### PR DESCRIPTION
**Pull request by mattradz**
Demos documentation, middle man demo (for a few quits)

### CODE REVIEW

File: docs/demos/quantum-teleportation.rst [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+            b = qubitsCharlie[0]
+
+            # Prepare State Based on Measurements
+            p.if_then(ro[0], X(b))
+            p.if_then(ro[1], Z(b))
+
+Create Program
+----------------------------------------
+We can now define our pyquil program to pass into each agent. For this demo, we will let qubit 2 be our state :math:`|\psi\rangle`, 
+prepared using a Hadamard. Therefore, we Alice has state :math:`|\psi\rangle = |2\rangle = \frac{1}{\sqrt{2}}(|0\rangle + |1\rangle)`.
```
Comment (zacespinosa) [inline docs/demos/quantum-teleportation.rst]: Therefore, Alice has state (omit "we")

File: docs/demos/quantum-teleportation.rst [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+is a method to deliver any arbitrary quantum state, :math:`|\psi\rangle`,
+between agents only using classical communication channels. This protocol allows Alice to deliver
+an unknown qubit state, :math:`|\psi\rangle`, to Bob using a shared bell state pair. 
+
+This protocol enables quantum computers to interact as a network in sharing arbitrary quantum states. 
+Note that due to the requirement of classical communication, this protocol does not validate faster-than-light
+communication. Quantum teleporatation transports a quantum state by sending two classical bits, and thus is the inverse
+of :ref:`superdense coding <superdense-coding>`.
+
+Source Code
```
Comment (zacespinosa) [inline docs/demos/quantum-teleportation.rst]: Thoughts on moving Source code section to the very end? I feel like most people would see this and just skip the entire tutorial if it's up here

File: docs/demos/quantum-teleportation.rst [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        # Create Bell State Pair
+        p = self.program
+        p += H(0)
+        p += CNOT(0,1)
+
+        self.qsend(alice.name, [0])
+        self.qsend(bob.name, [1])
+
+Now, we will create Agent Alice and Agent Bob. Alice will project her state :math:`|\psi\rangle = \alpha |0\rangle + \beta |1\rangle` onto her 
+Bell State Pair. Then, she will measure her two qubits, and send the results to Bob, which Bob will use to recreate Alice original state
```
Comment (zacespinosa) [inline docs/demos/quantum-teleportation.rst]: Alice's original state*

File: docs/demos/superdense-coding.rst [hunk 4]
```diff
@@ -0,0 +1,193 @@
+.. _superdense-coding: 
+
+=========================================================
+Superdense Coding
+=========================================================
+
+Superdense coding is a method to deliver any two classical bits using a single
+quantum bit. The protocol allows two agents, Alice and Bob, to use an entangled bell state pair
+along with interaction with a single quantum bit to transmit two classical bits of information .
```
Comment (zacespinosa) [inline docs/demos/superdense-coding.rst]: information. (remove space)

File: docs/demos/superdense-coding.rst [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+=========================================================
+Superdense Coding
+=========================================================
+
+Superdense coding is a method to deliver any two classical bits using a single
+quantum bit. The protocol allows two agents, Alice and Bob, to use an entangled bell state pair
+along with interaction with a single quantum bit to transmit two classical bits of information . 
+
+This protocol enables quantum computers to interact as a network in sharing classical information using quantum bits.
+Superdense coding transports two classical bits by sednding a quantum bit, and thus is the inverse
```
Comment (zacespinosa) [inline docs/demos/superdense-coding.rst]: sending*

File: docs/demos/superdense-coding.rst [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+of :ref:`quantum teleportation <quantum-teleportation>`.
+
+Source Code
```
Comment (zacespinosa) [inline docs/demos/superdense-coding.rst]: Same note, maybe move Source Code to end

File: docs/demos/superdense-coding.rst [hunk 7]
```diff
@@ -0,0 +1,193 @@
+.. _superdense-coding:
```
Comment (zacespinosa) [inline docs/demos/superdense-coding.rst]: is this file compiling correctly? For some reason I can't display the rich version

Comment (mattradz) [reply docs/demos/superdense-coding.rst]: It works when i call open _build/html/index.html

File: docs/demos/superdense-coding.rst (hunk 6 above)
Comment (mattradz) [reply docs/demos/superdense-coding.rst]: sg, moved

File: docs/demos/quantum-teleportation.rst (hunk 2 above)
Comment (mattradz) [reply docs/demos/quantum-teleportation.rst]: sg, moved

**Pull request merged**